### PR TITLE
dist/tools/doccheck: annotate errors in Github Action

### PR DESCRIPTION
**Pull request by miri64**
### Contribution description
Provides Github annotations from the `doccheck` script, using the harness provided in https://github.com/RIOT-OS/RIOT/pull/15640. It also changes the script not to early exit on error, but to accumulate the errors. This was mostly done so I don't have to teardown the annotation harness in multiple places, but I think, since the static tests run isolated anyways and `doxygen` takes ages to finish anyways, it does not hurt to run the checks after all the time as well.

### Testing procedure
I provided a test commit that will cause some annotations.

### Issues/PRs references
~~Depends on https://github.com/RIOT-OS/RIOT/pull/15640~~

### CODE REVIEW

**Comment by miri64** [comment]
Rebased to current master. This PR has no dependencies anymore, as #15640 was merged.

**Comment by miri64** [comment]
Ping?

File: dist/tools/doccheck/check.sh [hunk 1]
```diff
@@ -79,15 +98,22 @@ MULTIPLE_DEFINED_GROUPS=$(echo "${DEFINED_GROUPS}" | uniq -d)
...
-            grep "\<${group}\>$" | sort -u |
-            awk -F: '{ print "\t" $1 }';
+            grep "\<${group}\>$" | sort -u)
+        DEFGROUPFILES=$(echo "${DEFGROUPS}" | awk -F: '{ print "\t" $1 }')
+        echo -e "\n${CWARN}${group}${CRESET} defined in:";
+        echo "${DEFGROUPFILES}"
+        echo "${DEFGROUPS}" | while read defgroup;
+        do
+            github_annotate_error $(echo ${defgroup} | awk -F: '{ print $1,$2 }') \
+                "Multiple doxygen group define of '${group}' in\n${DEFGROUPFILES}"
```
Comment (aabadie) [inline dist/tools/doccheck/check.sh:114]:
```suggestion
                "Multiple doxygen group definitions of '${group}' in\n${DEFGROUPFILES}"
```
?

Comment (miri64) [reply dist/tools/doccheck/check.sh]: Used.

**Review by aabadie — APPROVED** [review]
ACK, looks good, please squash and remove the "REMOVE ME" commit.

**Comment by miri64** [comment]
Done!

**Comment by aabadie** [comment]
There's something I don't really like the annotation (maybe it was already reported in previous PRs): the regular check output is mixed with annotation output the action log window. So you get something like:

```
	ERROR There are 1 Doxygen groups defined multiple times:
	
	net_gnrc_sixlowpan defined in:
		sys/include/net/gnrc/ipv6.h
		sys/include/net/gnrc/sixlowpan.h
Error: unable to resolve reference to `net_gnrc_ipv6' for ref command
Error: unable to resolve reference to `net_gnrc_ipv6' for ref command
Error: Undefined doxygen group 'net_gnrc_ipv6'

Error: Multiple doxygen group definitions of 'net_gnrc_sixlowpan' in
	sys/include/net/gnrc/ipv6.h
	sys/include/net/gnrc/sixlowpan.h
Error: Undefined doxygen group 'net_gnr'
Error: Undefined doxygen group 'net_gnrc_ipv6'
Error: Undefined doxygen group 'net_gnrc_ipv6'
Error: Undefined doxygen group 'net_gnrc_ipv6'
Error: unable to resolve reference to `net_gnrc_ipv6' for ref command
Error: unable to resolve reference to `net_gnrc_ipv6' for ref command
Error: Undefined doxygen group 'net_gnrc_ipv6'
Error: Undefined doxygen group 'net_gnrc_ipv6'
Error: Undefined doxygen group 'net_gnrc_ipv6'
Error: Undefined doxygen group 'net_gnrc_ipv6'
Error: unable to resolve reference to `net_gnrc_ipv6' for ref command
Error: unable to resolve reference to `net_gnrc_ipv6' for ref command
Error: unable to resolve reference to `net_gnrc_ipv6' for ref command
Error: unable to resolve reference to `net_gnrc_ipv6' for ref command
Error: unable to resolve reference to `net_gnrc_ipv6' for ref command
Error: unable to resolve reference to `net_gnrc_ipv6' for ref command
Error: unable to resolve reference to `net_gnrc_ipv6' for ref command
Error: unable to resolve reference to `net_gnrc_ipv6' for ref command
Error: unable to resolve reference to `net_gnrc_ipv6' for ref command
Error: unable to resolve reference to `net_gnrc_ipv6' for ref command
Error: unable to resolve reference to `net_gnrc_ipv6' for ref command
Error: unable to resolve reference to `net_gnrc_ipv6' for ref command
Error: unable to resolve reference to `net_gnrc_ipv6' for ref command
Error: unable to resolve reference to `net_gnrc_ipv6' for ref command
Error: unable to resolve reference to `net_gnrc_ipv6' for ref command
Error: unable to resolve reference to `net_gnrc_ipv6' for ref command
Error: unable to resolve reference to `net_gnrc_ipv6' for ref command
Error: unable to resolve reference to `net_gnrc_ipv6' for ref command
Error: unable to resolve reference to `net_gnrc_ipv6' for ref command
Error: unable to resolve reference to `net_gnrc_ipv6' for ref command
Error: Multiple doxygen group definitions of 'net_gnrc_sixlowpan' in
	sys/include/net/gnrc/ipv6.h
	sys/include/net/gnrc/sixlowpan.h
Error: group net_gnrc_sixlowpan: ignoring title "6LoWPAN" that does not match old title "IPv6"
Error: unable to resolve reference to `net_gnrc_netap' for ref command
```

In the case of the doccheck, different types of error (non defined groups and groups defined multiple times) are mixed.

**Comment by miri64** [comment]
> There's something I don't really like with the annotation (maybe it was already reported in previous PRs): the regular check output is mixed with annotation output in the action log window. So you get something like:

So what should we do? Remove the regular output?

**Comment by aabadie** [comment]
> So what should we do? Remove the regular output?

I don't know, just noticed that :)
But yes, maybe the original check output could be skipped when launched via GH actions, to only keep annotation output. When run locally, print the original check output. Do you think it makes sense ?

**Comment by aabadie** [comment]
I think that for now we can go with this PR and later improve. The annotations are here to be inlined in the PR diff and people will most likely not look at the full job output. What do you think ?

**Comment by miri64** [comment]
I applied that change and returned the REMOVE ME commit to test.

File: sys/include/net/gnrc/ipv6.h [hunk 2]
```diff
@@ -7,11 +7,11 @@
  */
 
 /**
- * @defgroup    net_gnrc_ipv6 IPv6
- * @ingroup     net_gnrc
+ * @defgroup    net_gnrc_sixlowpan IPv6
```
Comment (miri64) [inline sys/include/net/gnrc/ipv6.h:10]: Mh.. this isn't parsed correctly. Will have a look.

**Comment by miri64** [comment]
Ping? It should work now.

**Comment by miri64** [comment]
Ok found some stuff still, but now it should work

**Comment by miri64** [comment]
Mhhh there is still something weird happening in the GitHub output I did not experience locally... will have to do some more debugging after all...

**Comment by miri64** [comment]
The check for if `echo -e` should be used was wrong. It should work now.

**Review by MrKevinWeiss — APPROVED** [review]
I can run it locally and it looks good on GH

<details>

```

ERROR There are 2 undefined Doxygen groups:

net_gnr found in:
	sys/include/net/gnrc/ipv6.h

net_gnrc_ipv6 found in:
	sys/include/net/gnrc/icmpv6.h
	sys/include/net/gnrc/ipv6/blacklist.h
	sys/include/net/gnrc/ipv6/ext.h
	sys/include/net/gnrc/ipv6.h
	sys/include/net/gnrc/ipv6/hdr.h
	sys/include/net/gnrc/ipv6/nib.h
	sys/include/net/gnrc/ipv6/whitelist.h
	sys/include/net/gnrc/ndp.h


ERROR There are 1 Doxygen groups defined multiple times:

net_gnrc_sixlowpan defined in:
	sys/include/net/gnrc/ipv6.h
	sys/include/net/gnrc/sixlowpan.h
```
</details>

ACK!

**Comment by MrKevinWeiss** [comment]
Please remove and squash (maybe we should add a tool to autoremove REMOVEME commits) :)

**Comment by miri64** [comment]
Done

**Pull request merged**